charts/ratify/templates/provider.yaml

```diff
@@ -1,18 +1,35 @@
+{{- $subject := printf "/CN=ratify.%s" .Release.Namespace -}}
+{{- $dns := printf "ratify.%s" .Release.Namespace -}}
+{{- $ca := genCA "/O=Ratify/CN=Ratify Root CA" 365 -}}
+{{- $cert := genSignedCert $subject nil (list $dns) 365 $ca -}}
+
+---
+
 {{ include "ratify.providerGKVersion" . }}
 kind: Provider
 metadata:
   name: ratify-provider
 spec:
   url: https://{{ include "ratify.fullname" .}}.{{ .Release.Namespace }}:6001/ratify/gatekeeper/v1/verify
   timeout: {{ required "You must provide .Values.provider.timeout.validationTimeoutSeconds" .Values.provider.timeout.validationTimeoutSeconds }}
-  caBundle: {{ required "You must provide .Values.provider.tls.cabundle" .Values.provider.tls.cabundle }}
+  {{ include "ratify.providerCabundle" (list . $ca) | nindent 2}}
+
 ---
+
 {{ include "ratify.providerGKVersion" . }}
 kind: Provider
 metadata:
   name: ratify-mutation-provider
 spec:
   url: https://{{ include "ratify.fullname" .}}.{{ .Release.Namespace }}:6001/ratify/gatekeeper/v1/mutate
   timeout: {{ required "You must provide .Values.provider.timeout.mutationTimeoutSeconds" .Values.provider.timeout.mutationTimeoutSeconds }}
-  caBundle: {{ required "You must provide .Values.provider.tls.cabundle" .Values.provider.tls.cabundle }}
-  
\ No newline at end of file
+  {{ include "ratify.providerCabundle" (list . $ca) | nindent 2}}
+
+---
+
+apiVersion: v1
```

Comment:
I know it's bad to move secret here, but providers and tls-secret need to share the `$ca` generated above. And if I move the top template to `_helpers.tpl`, a new `$ca` would be generated each time it's invoked. Just wonder if there is another option to make it work.

Reply:
yeah... its not ideal. I'm wondering if it's possible to run the scripts during the preinstall webhook and the load the files the install step via .Files reference like this: https://helm.sh/docs/chart_template_guide/accessing_files/

Reply:
@etrexel would you happen to have any ideas on how to get around this?

Reply:
@sozercan do you have any suggestion on it? You mentioned we can use helm hook to run a script. Does it work in this case?